delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/Resolvers.scala

```diff
@@ -28,19 +31,19 @@ trait Resolvers {
   /**
     * Create a new resolver where the id is either present on the payload or self generated
     * @param projectRef        the project where the resolver will belong
-    * @param resolverFields the payload to create the resolver
+    * @param payload the payload to create the resolver
     */
-  def create(projectRef: ProjectRef, resolverFields: ResolverFields)(implicit
+  def create(projectRef: ProjectRef, payload: Json)(implicit
```

Comment:
I would recommend to have both methods. Someone using the resolvers SDK that is not going through the API should not have to provide a json payload in order to create a resolver.

usually the `create (projectRef, json)` should then call the `create(projectRef, resolverFields)`

Reply:
Only the second create method would fit as we need an id.
And it is a little more complicated than that as we need to generate the json source from the value.
So I thought that as we don't need it for now, we could postpone their implementation

Reply:
Yes, we can postpone it.

Reply:
I think turning a json-ld document to a ResolverFields should be a serialization concern and it should not show in the contract. I would recommend the following signature(s):

```
def create(project: ProjectRef, id: Iri, value: ResolverValue, source: Json): ???
def create(project: ProjectRef, value: ResolverValue, source: Json): ???
...
```

Reply:
The thing is we need  a project to deserialize it properly and I follow the same pattern than the one with resources/schemas.
If we do it for resolvers, we should do it for the others no ?

Reply:
I don't see why this contract is better than the existing one. Serialization **is** a concern of the SDK since we need ids and other checks.

Reply:
I'm thinking about the default resolver that needs to be created for a project; I think it's nicer to have it as a value with specific fields rather than a json blob.

Reply:
Oh yes, I forgot that one.
So I will add now another method for it which takes a ResolverValue and generates the json source from it.

Reply:
And in the case of plugins, it's a lot simpler to deal with these value types. If we want to also expose the serialization part in the sdk, we can always add a new method and even possibly overload the others:

```
def valueFrom(source: Json): IO[ParsingRejection, (Option[Iri], ResolverValue)]
```